Userland/Libraries/LibGUI/AbstractView.cpp

```diff
@@ -72,6 +72,7 @@ void AbstractView::model_did_update(unsigned int flags)
         if (!model()->is_within_range(m_drop_candidate_index))
             m_drop_candidate_index = {};
         selection().remove_matching([this](auto& index) { return !model()->is_within_range(index); });
+        do_search(move(m_searching));
```

Comment:
Since this seems exceptionally weird at first glance, could we add a `find_next_search_match()` next to `do_search()` that doesn't touch `m_searching` and just finds the next match?
Assuming I'm right about what this intends to do anyway

Reply:
You're absolutely right. The `find_next_search_match()` is a good idea. Thanks!